### PR TITLE
add cloudfront module with example usage

### PR DESCRIPTION
**Pull request by mateuszwu**

### CODE REVIEW

File: examples/cloudfront/main.tf [hunk 1]
```diff
@@ -0,0 +1,84 @@
+resource "aws_s3_bucket" "example_s3_bucket" {
+  bucket = "tf-iam-example-s3-bucket"
+  acl    = "private"
+}
+
+resource "aws_cloudfront_origin_access_identity" "example_cloudfront" {
+}
+
+resource "aws_cloudfront_distribution" "example_cloudfront" {
```
Comment (qbart) [inline examples/cloudfront/main.tf]: rather than creating example from scratch, use our modules

File: modules/cloudfront/main.tf [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+      "cloudfront:CreateInvalidation",
+    ]
+
+    resources = [
+      var.cloudfront
+    ]
+  }
+}
+
+resource "aws_iam_policy" "this" {
```
Comment (qbart) [inline modules/cloudfront/main.tf]: missing `name` attribute

File: modules/cloudfront/main.tf [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  policy = data.aws_iam_policy_document.this.json
+}
+
+resource "aws_iam_policy_attachment" "this" {
+  name       = "cloudfront-invalidation-policy"
```
Comment (qbart) [inline modules/cloudfront/main.tf]:
```suggestion
  name       = "${var.name_prefix}-cdn-invalidation"
```

**Review by qbart — CHANGES_REQUESTED** [review]
small changes

File: modules/cloudfront/variables.tf [hunk 4]
```diff
@@ -0,0 +1,15 @@
+variable "name_prefix" {
+  type        = string
+  description = "Prefix that will be prepended to resources names"
```
Comment (qbart) [inline modules/cloudfront/variables.tf]:
```suggestion
  description = "Prefix that will be prepended to resource names"
```

File: examples/cloudfront/main.tf [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+      locations        = ["US", "CA", "GB", "DE", "PL"]
+    }
+  }
+
+  viewer_certificate {
+    cloudfront_default_certificate = true
+  }
+}
+
+output "CLOUDFRONT_domain_name" {
```
Comment (qbart) [inline examples/cloudfront/main.tf]:
```suggestion
output "cloudfront_domain_name" {
```

File: examples/cloudfront/main.tf [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+  value = aws_cloudfront_distribution.example_cloudfront.domain_name
+}
+
+output "CLOUDFRONT_distribution_id" {
```
Comment (qbart) [inline examples/cloudfront/main.tf]:
```suggestion
output "cloudfront_distribution_id" {
```

**Review by qbart — CHANGES_REQUESTED** [review]
few minors and module name could be named differently like cloudfront-invalidation

**Comment by mateuszwu** [comment]
> few minors and module name could be named differently like cloudfront-invalidation

Thanks, good idea 👍

**Review by qbart — APPROVED** [review]

**Pull request merged**